rpc/client.go

```diff
@@ -106,15 +107,38 @@ func NewClient(addr net.Addr, opts *retry.Options, context *Context) *Client {
 		Closed:       make(chan struct{}),
 		clock:        context.localClock,
 		remoteClocks: context.RemoteClocks,
-		cached:       !context.DisableCache,
 	}
-	if !context.DisableCache {
-		clients[c.Addr().String()] = c
+	clients[c.Addr().String()] = c
+
+	tlsConfig, tlsErr := context.GetClientTLSConfig()
```

Comment:
I would actually pass the context to the dialer, and have those call GetClientTLSConfig. Also, calling it more than once doesn't matter, it does lazy loading of the actual configs, so everything after the first successful call is cheap.